### PR TITLE
Partition write support new collations

### PR DESCRIPTION
**Pull request by xuanyu66**
### What problem does this PR solve? 

In https://github.com/pingcap/tispark/pull/2524, TiSpark supports new collation.
According to [Range column partition](https://dev.mysql.com/doc/refman/5.7/en/partitioning-columns-range.html#:~:text=Range%20columns%20partitioning%20is%20similar,types%20other%20than%20integer%20types), partitioning also needs to obey the sort orders of collation.

### What is changed and how it works?

- For range column partition,  using different sort orders according to collation

### Check List 

Tests 

 - Integration test

Related changes

 - Need to update the documentation
 - Need to be included in the release note

### CODE REVIEW

File: tikv-client/src/main/java/com/pingcap/tikv/expression/visitor/RangePartitionLocator.java [hunk 1]
```diff
@@ -59,6 +66,7 @@ public Boolean visit(ComparisonBinaryExpression node, PartitionLocatorContext co
         Expression expression = left.getChildren().get(0);
         ColumnRef columnRef = (ColumnRef) expression;
         columnRef.resolve(tableInfo);
+        dataType = IntegerType.BIGINT;
```
Comment (Daemonxiao) [inline tikv-client/src/main/java/com/pingcap/tikv/expression/visitor/RangePartitionLocator.java:69]: Why set YEAR as IntergerType? Shall we add an annotation here?

Comment (xuanyu66) [reply tikv-client/src/main/java/com/pingcap/tikv/expression/visitor/RangePartitionLocator.java:69]: function year() return integer

**Review by shiyuhang0 — APPROVED** [review]

**Review by Daemonxiao — APPROVED** [review]

**Pull request merged**